pandas/sparse/array.py

```diff
@@ -152,9 +152,17 @@ def __new__(cls, data, sparse_index=None, index=None, kind='integer',
 
         # Create array, do *not* copy data by default
         if copy:
-            subarr = np.array(values, dtype=dtype, copy=True)
+            try:
+                # ToDo: Can remove this error handling when we actually
+                # support other dtypes
```

Comment:
Can I leave this ATM, because this can be removed entirely after changing ``dtype`` default to ``None`` (#667).